### PR TITLE
8322816: RISC-V: Incorrect guarantee in patch_vtype 

### PR DESCRIPTION
**Pull request by kimbarrett**
Please review this change that fixes a test for a guarantee.  This also
removes a -Wparentheses warning when those are enabled (which is how the
problem was discovered).

The problem is that operator precedence groups the sub-expressions differently
than intended. The fix is to override the operator precedence by adding
parentheses to achieve the intended grouping.

Testing: Local (linux-x64) cross-build for linux-riscv with this change plus
-Wparentheses enabled and other changes to allow that to work.

Requesting someone from the riscv porters to properly test this.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8322816](https://bugs.openjdk.org/browse/JDK-8322816): RISC-V: Incorrect guarantee in patch_vtype (**Bug** - P4)

### Reviewers
 * [Fei Yang](https://openjdk.org/census#fyang) (@RealFYang - **Reviewer**)
 * [Ludovic Henry](https://openjdk.org/census#luhenry) (@luhenry - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/17215/head:pull/17215` \
`$ git checkout pull/17215`

Update a local copy of the PR: \
`$ git checkout pull/17215` \
`$ git pull https://git.openjdk.org/jdk.git pull/17215/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 17215`

View PR using the GUI difftool: \
`$ git pr show -t 17215`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/17215.diff">https://git.openjdk.org/jdk/pull/17215.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/17215#issuecomment-1873689393)

### CODE REVIEW

File: src/hotspot/cpu/riscv/assembler_riscv.hpp
Comment (RealFYang) [inline src/hotspot/cpu/riscv/assembler_riscv.hpp:1160]: I see the `vill` parameter is always false in current code, which means this guarantee never gets excecuted. And I don't think we would make use of the `vill` field of vtype in future. So I personally perfer to remove this guarantee and its enclosing if block for now.

Comment (kimbarrett) [reply src/hotspot/cpu/riscv/assembler_riscv.hpp:1160]: Rather than removing the guarantee, wouldn't it be better to guarantee/assert `vill == 0`?
Although looking at uses, that argument is a bool, so it should be `guarantee(!vill, ...)`.

Comment (RealFYang) [reply src/hotspot/cpu/riscv/assembler_riscv.hpp:1160]: Hi, Yes, that's better. Maybe: `guarantee(!vill, "should be");`

Comment (kimbarrett) [reply src/hotspot/cpu/riscv/assembler_riscv.hpp]: I've changed the guarantee as discussed.  There are further cleanups possible
here, but I'll leave that to the riscv port maintainers.

**Review by RealFYang — APPROVED** [review]

**Comment by kimbarrett** [comment]
Waiting for second review.

**Review by luhenry — APPROVED** [review]

**Pull request closed without merging**
